whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java

```diff
@@ -1938,6 +1962,105 @@ public void search_inetnum_with_resource_flag() {
         assertTrue(hasSourceTest);
     }
 
+    @Test
+    public void lookup_person_with_proxy_not_allowed() {
+
+        databaseHelper.addObject("" +
+                "person:    Lo Person\n" +
+                "admin-c:   TP1-TEST\n" +
+                "tech-c:    TP1-TEST\n" +
+                "nic-hdl:   LP1-TEST\n" +
+                "mnt-by:    OWNER-MNT\n" +
+                "source:    TEST\n");
+
+        try {
+            RestTest.target(getPort(), "whois/search?query-string=LP1-TEST&source=TEST&client-attribute=testId,10.1.2.3")
+                    .request(MediaType.APPLICATION_XML)
+                    .get(WhoisResources.class);
+            fail();
+        } catch (WebApplicationException e) {
+            assertThat(e.getResponse().getStatus(), is(400));
+            assertThat(e.getResponse().readEntity(String.class), containsString("ERROR:203: you are not allowed to act as a proxy"));
+        }
+    }
+
+    @Test
+    public void lookup_person_with_client_flag_no_proxy() throws Exception {
+        final InetAddress localhost = InetAddress.getByName(LOCALHOST);
+
+        databaseHelper.addObject("" +
+                "person:    Lo Person\n" +
+                "admin-c:   TP1-TEST\n" +
+                "tech-c:    TP1-TEST\n" +
+                "nic-hdl:   LP1-TEST\n" +
+                "mnt-by:    OWNER-MNT\n" +
+                "source:    TEST\n");
+
+        try {
+            final int limit = accessControlListManager.getPersonalObjects(localhost);
+
+            final WhoisResources whoisResources = RestTest.target(getPort(), "whois/search?query-string=LP1-TEST&source=TEST&flags=no-filtering&flags=rB&client-attribute=testId")
+                    .request(MediaType.APPLICATION_XML)
+                    .get(WhoisResources.class);
+
+            assertThat(whoisResources.getErrorMessages(), is(empty()));
+            assertThat(whoisResources.getWhoisObjects(), hasSize(1));
+
+            //ACL is accounted for as there is no proxy ip specified
+            final int remaining = accessControlListManager.getPersonalObjects(localhost);
+            assertThat(remaining, is(limit - 1));
+
+            assertThat(whoisResources.getParameters().getClientAttribute(), is("testId"));
+
+        } finally {
+            testPersonalObjectAccounting.resetAccounting();
```

Comment:
finally block not necessary, if these statements are also in the "After" method?

Reply:
agree